scripts/add_existing_baseyear.py

```diff
@@ -623,6 +624,226 @@ def add_heating_capacities_installed_before_baseyear(
+    start_dates_bof = round(start_dates["Integrated steelworks"])
+
+    # Average age of assets in Iron and steel in Europe: 21-28 years, so I assume they are starting in 2000 in case https://www.energimyndigheten.se/4a9556/globalassets/energieffektivisering_/jag-ar-saljare-eller-tillverkare/dokument/produkter-med-krav/ugnar-industriella-och-laboratorie/annex-b_lifetime_energy.pdf
+    start_dates_eaf = start_dates_eaf.where((start_dates_eaf >= 1000) & np.isfinite(start_dates_eaf), 2000)
+    start_dates_bof = start_dates_bof.where((start_dates_bof >= 1000) & np.isfinite(start_dates_bof), 2000)
+
+    nodes = pop_layout.index
+    p_nom_bof = pd.DataFrame(index=nodes, columns=(["value"]))
+    p_nom_eaf = pd.DataFrame(index=nodes, columns=(["value"]))
+
+    p_nom_bof = capacities_bof / nhours  # get the hourly production capacity
+    p_nom_eaf = capacities_eaf / nhours  # get the hourly production capacity
+
+    # Should steel be produced at a constant rate during the year or not? 1 or 0
+    prod_constantly = 0
+    ramp_limit = 0
+
+    ########### Add existing steel production capacities ############
+    # Blast furnace assuming with natural gas
+
+    # BOF
+
+    iron_to_steel_bof = 1.429
+    iron_to_steel_eaf_ng = 1.36
+    em_factor_bof = 668.2 # tCO2/kt steel
+
+    # Lifetimes https://www.energimyndigheten.se/4a9556/globalassets/energieffektivisering_/jag-ar-saljare-eller-tillverkare/dokument/produkter-med-krav/ugnar-industriella-och-laboratorie/annex-b_lifetime_energy.pdf
+    lifetime_bof = 100
+    lifetime_eaf = 67 
+
+    # Capex
+    discount_rate = 0.04
+    capex_bof = ((211000 + 100000 ) * 0.7551 / nhours / iron_to_steel_bof ) * calculate_annuity(lifetime_bof, discount_rate)
+    # https://iea-etsap.org/E-TechDS/PDF/I02-Iron&Steel-GS-AD-gct.pdf 2010USD/kt/yr steel, then /nhour for the price in 2010USD/kt steel/timestep, divided by the efficiency to have the value in kt iron
+    capex_eaf = ((145000 + 80000) * 0.7551 / nhours / iron_to_steel_bof) * calculate_annuity(lifetime_eaf, discount_rate)
+    # https://iea-etsap.org/E-TechDS/PDF/I02-Iron&Steel-GS-AD-gct.pdf then /nhours for the price
+    opex_bof = 90 * 1e3 * 0.7551 / nhours / iron_to_steel_bof # $/t/yr -> €/kt iron/hour
+    opex_eaf = (13+32) * 1e3 * 0.7551 / nhours / iron_to_steel_bof # $/t/yr -> €/kt iron/hour
+
+    # Energy input data BF BOF
+    coal4bfobf = 4415.6 # MWh coal per kt steel
+    coal4bfbof_harvey = 1306 + 130 # MWh coal per kt steel
+
+    elec4bfbof = 488.9 # MWh electricity per kt steel
+    elec4bfbof_harvey = 75 + 303 # MWh electricity per kt steel
+
+    # Energy input data NG DRI EAF
+    ng4dri = 2803 # MWh natural gas per kt steel
+    ng4dri_harvey = 2733 # MWh natural gas per kt steel
+
+    hy4dri = 2211 # MWh natural gas per kt steel
+
+    elec4eaf = 675.9 # MWh electricity per kt steel
+    elec4eaf_harvey = 588 # MWh electricity per kt steel
+
+
+    n.add(
+        "Link",
+        nodes,
+        suffix=" BF-BOF-2020",
```

Comment:
There should be a `logger.error` if the year is not 2020
Also: there could be steel plants in the database that are built after 2020 and their capacity would be added here, right?
We discussed this multiple time, what it means to optimize a year (e.g. 2020 and 2030) if our optimization represents then the middle (2025) or the beginning of the span (2020).